server/protocols/webadmin/webadmin-mailrepository/src/main/java/org/apache/james/webadmin/dto/MailDto.java

```diff
@@ -172,8 +171,8 @@ private static HeadersDto fetchPerRecipientHeader(
             return Optional.empty();
         }
 
-        return Optional.of(Iterators.toStream(mail.getAttributeNames())
-            .collect(Guavate.toImmutableMap(Function.identity(), attributeName -> mail.getAttribute(attributeName).toString())));
+        return Optional.of(Iterators.toStream(mail.attributeNames())
+            .collect(Guavate.toImmutableMap(AttributeName::asString, attributeName -> mail.getAttribute(attributeName).toString())));
```

Comment:
ImmutableMap.toImmutableMap

Reply:
Is Guavate deprecated?

Reply:
https://github.com/steveash/guavate

`As of 2017-01-12 Guava 21 was released which brings Java 8 support to the library. That removes the need for Guavate. Please upgrade to Guava 21`